joblib/_store_backends.py

```diff
@@ -30,8 +31,11 @@ class CacheWarning(Warning):
 
 def concurrency_safe_write(object_to_write, filename, write_func):
     """Writes an object into a unique file in a concurrency-safe way."""
+    # UUID is unique across node and time,
+    # Add thread_id and process_id to avoid collisions due to concurrency
```

Comment:
```suggestion
    # Add thread_id and process_id to improve debugability.
```

Reply:
It is not really for debugability, more because uuid is not different for different process/thread if it is called at the same time.